cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/cube/CubeQuery.java

```diff
@@ -45,29 +45,29 @@
   private final long endTs;
   private final int resolution;
   private final int limit;
-  private final String measureName;
+  private final List<String> measureNames;
   private final MeasureType measureType;
   private final Map<String, String> sliceByTagValues;
   private final List<String> groupByTags;
   private final Interpolator interpolator;
 
   public CubeQuery(long startTs, long endTs, int resolution, int limit,
```

Comment:
You should add a contructor that takes in one metricName, not list - that would simplify a lot of client code (you probably saw that in tests you changes). Esp. for Cube dataset users that would be not better